core/src/main/scala/kafka/log/FileMessageSet.scala

```diff
@@ -149,6 +155,56 @@ class FileMessageSet private[kafka](@volatile var file: File,
   }
 
   /**
+   * Search forward for the message whose timestamp is greater than or equals to the target timestamp.
+   * The method will return
+   * <ul>
+   * <li>None if there is no message after the starting position.
+   * <li>TimestampOffset( largest_timestamp_checked_before_target_timestamp, the_next_offset_to_read ) otherwise.
+   * </ul>
+   *
+   * The search will stop immediately when it sees a message in format version before 0.10.0. This is to avoid
+   * scanning the entire log when all the messages are still in old format.
+   *
+   * @param targetTimestamp The timestamp to search for.
+   * @param startingPosition The starting position to search.
+   * @return None, if no message exists after starting position.
+   *         Some(TimestampOffset) with (largest_timestamp_checked, the_next_offset_to_read) otherwise.
```

Comment:
The comments of the return value are inconsistent. largest_timestamp_checked_before_target_timestamp and largest_timestamp_checked.